core/test/scripts/Voting.js

```diff
@@ -236,48 +297,116 @@ contract("scripts/Voting.js", function(accounts) {
+    assert.equal(
+      (await voting.getPrice(identifier, time)).toString(),
+      TEST_IDENTIFIERS["Custom Index (1)"].expectedPrice
+    );
   });
 
   it("Numerator/Denominator", async function() {
-    const identifier = web3.utils.utf8ToHex("0.5");
-    const time = "1560762000";
-
-    VotingScript.SUPPORTED_IDENTIFIERS["0.5"] = {
-      numerator: {
-        dataSource: "Constant",
-        value: "1"
-      },
-      denominator: {
-        dataSource: "Constant",
-        value: "2"
-      }
-    };
+    const identifier = TEST_IDENTIFIERS["0.5"].key;
+    const time = await voting.getCurrentTime();
+
     // Request an Oracle price.
-    await supportedIdentifiers.addSupportedIdentifier(identifier);
     await voting.requestPrice(identifier, time);
 
     const votingSystem = new VotingScript.VotingSystem(voting, voter, [new MockNotifier()]);
     await moveToNextRound(voting);
-    await votingSystem.runIteration();
+    let result = await votingSystem.runIteration();
+    assert.equal(result.batches, 1);
+    assert.equal(result.updates.length, 1);
+    assert.equal(result.skipped.length, 0);
+    assert.equal(result.failures.length, 0);
     await moveToNextPhase(voting);
-    await votingSystem.runIteration();
+    result = await votingSystem.runIteration();
+    assert.equal(result.batches, 1);
+    assert.equal(result.updates.length, 1);
+    await moveToNextRound(voting);
+
+    assert.equal((await voting.getPrice(identifier, time)).toString(), TEST_IDENTIFIERS["0.5"].expectedPrice);
+  });
+
+  it("Only batches up to the maximum number of commits or reveals that can fit in one block", async function() {
+    const identifier = TEST_IDENTIFIERS["Custom Index (100)"].key;
+    let testTransactions = Math.round(VotingScript.BATCH_MAX_COMMITS * 2 + 1);
+    const time = (parseInt((await voting.getCurrentTime()).toString()) - testTransactions).toString();
+
+    // Request Oracle prices.
+    for (i = 0; i < testTransactions; i++) {
+      let timeToVote = parseInt(time) + i;
+      await voting.requestPrice(identifier, timeToVote.toString());
+    }
+
+    // Sanity check.
+    assert.isFalse(await voting.hasPrice(identifier, time));
+
+    // Query pending requests to vote on during each phase.
+    let pendingVotes = await voting.getPendingRequests();
+    assert.equal(pendingVotes.length, 0, "There should be 0 pending requests during pre-commit phase");
+    const votingSystem = new VotingScript.VotingSystem(voting, voter, [new MockNotifier()]);
+
+    // Move to commit phase.
+    await moveToNextRound(voting);
+    pendingVotes = await voting.getPendingRequests();
+    assert.equal(
+      pendingVotes.length,
+      testTransactions,
+      `There should be ${testTransactions} pending requests during commit phase`
+    );
+    let result = await votingSystem.runIteration();
+    batchesExpected = Math.floor(testTransactions / VotingScript.BATCH_MAX_COMMITS);
+    if (testTransactions % VotingScript.BATCH_MAX_COMMITS > 0) {
```

Comment:
Maybe I'm wrong, but could you just do `Math.ceil()` instead of `Math.floor()` on the line above to avoid this `if` statement entirely?

Reply:
yep